### DISCUSSION TITLE
Fortran compile issues on MacOS Big Sur

### DISCUSSION THREAD

**Discussion by drsteve** [spacepy/spacepy#583] [Support and Q&A] [answered]
Not me, but sharing on behalf of a user asking me questions.

Source code obtained from github, and building using `setup.py`
The issue is that at some point in the IRBEM compilation `ld` complains about not being able to find `-lSystem`.

My thought was that perhaps xcode CLI tools weren't installed, but `xcode-select --install` says that command line tools are already present.
Log:
[diagnostics.txt](https://github.com/spacepy/spacepy/files/8226760/diagnostics.txt)

**Comment by jtniehof** [comment]
This is happening before we even start compiling irbem; it's in the f2py call to generate a signature file. It's cute that we then go on to try and build irbem anyhow, so I need to fix that. (Apparently f2py doesn't return a non-zero exit code in that case.) So this is almost certainly a numpy/f2py issue and, on our side, it would be a documentation issue. Kneejerk reaction is to make sure numpy is up to date. I can potentially dig into my mac later and see if there may be more subtle issues.

**Reply by drsteve** [reply]
Numpy is 1.20.3

I suggested (following https://stackoverflow.com/questions/56156520/gfortran-error-ld-library-not-found-for-lsystem-when-trying-to-compile) that `LIBRARY_PATH` was set to include `/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/lib`. Compilation success! Calls to `irbempy` functions post-install work!

Because I was debugging via email I had them set both `LIBRARY_PATH` and `LD_LIBRARY_PATH`. I'm not sure whether `LD_LIBRARY_PATH` will need to be set in the .bashrc or .zshrc (depending on which shell is being used) for future runtime path resolution...

**Comment by spacecataz** [comment]
Some anecdotal feedback on this, which I have run into before in different forms.
If you are upgrading from old versions of xcode (i.e., if your BigSur installation is an upgrade from a previous OS version), Apple moved things around which broke a lot of linking. At one point, I had to uninstall and reinstall Xcode (basically starting from scratch with my software install via macports) to overcome this issue. If you're seeing this issue with other compiles, too (not just Spacepy/Irbem), you may consider this path if you've run out of options.

**Reply by jtniehof** [reply]
Thanks. I'll capture this when I update the Mac install docs.

**Reply by jtniehof** [reply]
FWIW this is showing up for me even with a complete xcode uninstall and reinstall, so going to be putting in some installer code to peanut butter over it.

**Comment by jtniehof** [comment, marked as answer]
0.4.0 (via #639) both catches if f2py fails and puts the directory with -lSystem in the linker path. I've confirmed that simply installing xcode fresh does not fix the link path issue. I'm going to lock this discussion because, if those changes did not fix the issue, they're likely to be unrelated, and this might just muddy the waters.
